src/components/tooltip/Tooltip.tsx

```diff
@@ -217,8 +219,24 @@ class Tooltip extends React.Component<TooltipProps, State> {
             stopBubble,
             text,
             theme,
+            intl,
         } = this.props;
 
+        let ariaLabel;
+        let tooltipText;
+
+        if (text) {
+            if (typeof text === 'string') {
+                tooltipText = text;
+            } else if (React.isValidElement(text) && getProp(text, 'props.id')) {
```

Comment:
Can you check for the element type being FormattedMessage (because anything can have an `id`), and pull out the exact react-intl props here and when you pass them into formatMessage

```js
const { id, defaultMessage, description, values } = props;
intl.formatMessage({ id, defaultMessage, description }, values)
```

Reply:
cc @megansmith-box @DanDeMicco 👀 can you verify if this approach conflicts with any of the ways product adoption tooltips work? Do we put rich text formatting in there?

Reply:
@alexkrolick do you mean something like this?
```
      if (text) {
          if (typeof text === 'string') {
              tooltipText = text;
          } else if (React.isValidElement(text) && getProp(text, 'props')) {
              const { id, defaultMessage, description, values } = getProp(text, 'props');
              if (id && defaultMessage && description) {
                  tooltipText = intl.formatMessage({ id, defaultMessage, description }, values)
              }
          }
      }
```

Reply:
Yes, like that